### PR TITLE
[POC] new generateSDL Gradle task

### PR DESCRIPTION
**Pull request by dariuszkuc**
Creating a new generic `graphqlGenerateSDL` Gradle task that can generate GraphQL schema in SDL format at built time. New task utilizes Gradle [Worker API](https://docs.gradle.org/nightly/userguide/worker_api.html) to isolate the plugin classpath from the build classpath.

In order to generate the SDL we rely on the [ServiceLoader](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ServiceLoader.html) mechanism to locate an instance of a `SchemaGeneratorHooksProvider` that is responsible for creating an instance of the hooks (default no-op hooks and federated hooks are provided by the `graphql-kotlin` lib).

```kotlin
interface SchemaGeneratorHooksProvider {
    fun create(): SchemaGeneratorHooks
}
```

With the change to `Worker API` we can update our plugin directory structure to the following

```
plugins
|- graphql-kotlin-gradle-plugin
|- graphql-kotlin-maven-plugin
|- client
  |- graphql-kotlin-client-generator // current plugin-core
|- schema
  |- graphql-kotlin-default-hooks-provider
  |- graphql-kotlin-sdl-printer
  |- graphql-kotlin-hooks-provider
  |- graphql-kotlin-federated-hooks-provider
```

---
Within corresponding Maven MOJO I'm thinking we would invoke the SDL generation process (that still uses the service loader mechanism and explicit plugin dependencies for the hooks providers) but we would need to explicitly provide target classpath.

---
resolves: https://github.com/ExpediaGroup/graphql-kotlin/issues/705

### CODE REVIEW

**Comment by smyrick** [comment]
Do we need separate packages for all the different hook providers? Can we just have one package for those and one for the  schema SDL printer plugin?

File: graphql-kotlin-types/src/main/kotlin/com/expediagroup/graphql/operations/Query.kt [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package com.expediagroup.graphql.spring.operations
+package com.expediagroup.graphql.operations
```
Comment (smyrick) [inline graphql-kotlin-types/src/main/kotlin/com/expediagroup/graphql/operations/Query.kt:17]: Should we create a separate `com.expediagroup.graphql.server` package as this marker interfaces are only used for the server?

Comment (dariuszkuc) [reply graphql-kotlin-types/src/main/kotlin/com/expediagroup/graphql/operations/Query.kt:17]: 👍 sure, I'll move this change into separate PR.

**Comment by dariuszkuc** [comment]
> Do we need separate packages for all the different hook providers? Can we just have one package for those and one for the schema SDL printer plugin?

We can but I think it complicates things. I kept it simple - i.e. single jar = single provider so we 

1) Don't need to add some ordering logic to determine which provider should be picked. (assuming single provider jar that has interface + default implementations) 
2) Don't need custom logic to determine whether to generate federated or non federated schema - its determined by the provided hook instance. We could add additional parameter for switching between federation/default but that flag would be ignored for custom providers.

**Comment by dariuszkuc** [comment]
Split up into:
* SDL generator PR: https://github.com/ExpediaGroup/graphql-kotlin/pull/990
* Gradle task PR: https://github.com/ExpediaGroup/graphql-kotlin/pull/993
* Maven mojo PR: https://github.com/ExpediaGroup/graphql-kotlin/pull/994

**Pull request closed without merging**